### ISSUE TITLE
Role doesn't exist but should

### ISSUE THREAD

**Issue by feuGeneA** [brianc/node-postgres#1647]
I found a few other (closed) issues describing this problem, but none described a solution.

pg.js:
```
const express = require('express');
const app = express();
const { Client } = require('pg');
const client = new Client({user: 'postgres'});
client.connect().catch((error)=>{console.log(error)});
```
```
$ node pg.js
error: role "postgres" does not exist
$ sudo -u postgres psql -c "\du;"
                             List of roles
 Role name |                   Attributes                   | Member of 
-----------+------------------------------------------------+-----------
 postgres  | Superuser, Create role, Create DB, Replication | {}
```

#868 suggested trying environment variables instead of programmatic configuration, but that didn't seem to work for me either:

pg2.js:
```
const express = require('express');
const app = express();
const { Client } = require('pg');
const client = new Client();
client.connect().catch((error)=>{console.log(error)});
```
```
$ PGUSER=postgres node pg.js
error: role "postgres" does not exist
```

My `package.json`  lists `"pg": "^7.4.3"`, and I'm using PostgreSQL v9.3.22, installed via the debian package in the repo for my Ubuntu 14.04.1 system.

What other info can I gather to diagnose this?

**Comment by charmander** [comment]
Does `psql -h 127.0.0.1 -U postgres` give the same error? pg’s default is to use 127.0.0.1, but psql’s default is a Unix domain socket whose path is set at compile time.

Try also the other way around:

```javascript
const client = new Client({
    host: '/var/run/postgresql',
    user: 'postgres',
});
```

(I don’t know if `/var/run/postgresql` is right for Ubuntu 14.04; substitute the real path to `.s.PGSQL.5431`.) Maybe even as the `postgres` user.

**Comment by feuGeneA** [comment]
I did get the same error with the command you mentioned, so it's clear this isn't a problem in this repo, so I'm closing this issue.

(In case anyone cares: `var/run/postgresql` was listed as the default in `psql --help`, but interestingly I didn't see `.s.PGSQL.5431` listed there; rather, only a `.s.PGSQL.5433`.  Digging a little more, I found a `...5432` socket in `/tmp`, and furthermore I found that there are two different postgres processes running these sockets.  Maybe I installed postgres at some point in the past using /tmp and then recently installed a different build that's using `/var/run/postgresql`?  I tried the client object configuration as suggested, with both sockets I found (and with corresponding port numbers), and I got varying authentication/access errors.  Whatever...)

I spun up a pg instance on AWS and I'm able to connect to that just fine.

Sorry for the noise, and thank you for helping me troubleshoot.
